beacon_node/beacon_chain/src/historical_blocks.rs

```diff
@@ -44,7 +52,134 @@ impl From<StoreError> for HistoricalBlockError {
     }
 }
 
+impl From<SignatureSetError> for HistoricalBlockError {
+    fn from(err: SignatureSetError) -> Self {
+        match err {
+            // The encoding of the signature is invalid, peer fault
+            e @ SignatureSetError::SignatureInvalid(_) => Self::InvalidSignature(format!("{e:?}")),
+            // All these variants are internal errors or unreachable for historical block paths,
+            // which only check the proposer signature.
+            // BadBlsBytes = Unreachable
+            e @ (SignatureSetError::BeaconStateError(_)
+            | SignatureSetError::ValidatorUnknown(_)
+            | SignatureSetError::ValidatorPubkeyUnknown(_)
+            | SignatureSetError::IncorrectBlockProposer { .. }
+            | SignatureSetError::MismatchedPublicKeyLen { .. }
+            | SignatureSetError::PublicKeyDecompressionFailed
+            | SignatureSetError::BadBlsBytes { .. }
+            | SignatureSetError::InconsistentBlockFork(_)) => Self::Unexpected(format!("{e:?}")),
+        }
+    }
+}
+
+impl From<AvailabilityCheckError> for HistoricalBlockError {
+    fn from(e: AvailabilityCheckError) -> Self {
+        Self::AvailabilityCheckError(e)
+    }
+}
+
 impl<T: BeaconChainTypes> BeaconChain<T> {
+    fn verify_blobs_and_data_columns(
+        &self,
+        blocks: &Vec<RpcBlock<T::EthSpec>>,
+    ) -> Result<Vec<AvailableBlock<T::EthSpec>>, HistoricalBlockError> {
+        // Verify that blobs or data columns signatures match
+        let sig_timer = metrics::start_timer(&metrics::BACKFILL_SIGNATURE_TOTAL_TIMES);
+        let setup_timer = metrics::start_timer(&metrics::BACKFILL_SIGNATURE_SETUP_TIMES);
+        // TODO: this logic is redundant with one from range sync. Do we have a good place to make
+        // it as a common function?
+        blocks
+            .iter()
+            .map(|block| {
+                if let Some(indices) = block.non_matching_blobs_signed_headers() {
+                    if !indices.is_empty() {
+                        return Err(HistoricalBlockError::InvalidBlobsSignature(indices));
+                    }
+                }
+                if let Some(indices) = block.non_matching_custody_columns_signed_headers() {
+                    if !indices.is_empty() {
+                        return Err(HistoricalBlockError::InvalidDataColumnsSignature(indices));
+                    }
+                }
+                Ok(())
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+        drop(setup_timer);
```

Comment:
No need to time this it should be quasi instant, `BACKFILL_SIGNATURE_VERIFY_TIMES` should be enough